src/Service.Tests/Unittests/RuntimeConfigLoaderJsonDeserializerTests.cs

```diff
@@ -128,6 +128,51 @@ public void CheckCommentParsingInConfigFile()
             Assert.IsTrue(RuntimeConfigLoader.TryParseConfig(actualJson, out RuntimeConfig _), "Should not fail to parse with comments");
         }
 
+        /// <summary>
+        /// Test to validate that optional properties
+        /// are nullable, don't contain defaults on serialization
+        /// but have the effect of default values when deserialized.
+        /// It starts with a minimal config and incrementally
+        /// adds the optional subproperties. At each step, tests for valid deserialization.
+        [TestMethod]
+        public void TestNullableOptionalProps()
+        {
+            // Test with no runtime property
+            StringBuilder minJson = new(@"
+                                ""data-source"": {
+                                    ""database-type"": ""mssql"",
+                                    ""connection-string"": ""@env('test-connection-string')""
+                                    },
+                                ""entities"": { }");
+
+            TryParseAndAssertOnDefaults("{" + minJson + "}", out _);
+
+            // Test with an empty runtime property
+            minJson.Append(@", ""runtime"": ");
+            string emptyRuntime = minJson + "{ }}";
```

Comment:
```suggestion
            minJson.Append(@", ""runtime"": { } ");
            string emptyRuntime = minJson + "}";
```
Would adding the empty runtime object to the first line work? It's more readable that way.

Reply:
meant to delete this comment, this is not possible after looking at logic in the lines that follow.

Reply:
I agree thats more readable - but Im building up the minJson by incrementally appending to it, so on line 156 - I add the sub properties to the `runtime` object. If I append the { } to `minJson`, would need to remove those first before adding the subproperties - wanted to avoid that.

Reply:
Understood